### ISSUE TITLE
Wrong index/order of blend shapes when importing a Blender mesh

### ISSUE THREAD

**Issue by Musashi98** [godotengine/godot#85293]
### Godot version

4.1.3.stable

### System information

Windows 10

### Issue description

When you import a model from a normal .blend file (I think this is in newer versions of Godot) that has shape keys (Blend Shapes property in the model mesh), the order of the Blend Shapes don't match with the indexes to change them through a script. In the GDScript you normally would use set_blend_shape_value() and you pass the index, but the index you pass would refer to the index the shape key was in blender and not the listed in the Blend Shapes property in the Inspector in Godot, it is very confusing. In Godot the Blend Shapes are ordered alphabetically by name, but in Blender they are ordered as they are created. It can be solved by rearrange the shape keys of the model in blender alphabetically by name, but is kind of boring work and can be forgoten. They should be appearing in the Inspector in Godot in the order of the indexes they have in Blender or change the indexes when the import happens so they match the order in the Blend Shapes property.

### Steps to reproduce

Create a simple model in Blender with two shape keys, but the first shape key must have a name bigger alphabetically than the second. Then import it to a project in Godot. There they will appear in the Blend Shapes in the wrong order (alphabetically is best order but the indexes don't match). Then in a GDScript try to change the value of the first Blend shape by doing set_blend_shape_value(0, something), the second Blend Shape will change instead of the first one.

### Minimal reproduction project

![Blender](https://github.com/godotengine/godot/assets/90828355/57103907-d493-4700-a2e6-367c1e95af80)
![Godot](https://github.com/godotengine/godot/assets/90828355/68397cec-5bcb-4b0a-a647-a0bf52c1de3c)

**Comment by RedMser** [comment]
It seems to intentionally sort alphabetically in the code. Though not sure if it's to workaround the non-deterministic ordering issues that may come with a HashMap, or if it's intended to be alphabetical...

https://github.com/godotengine/godot/blob/master/scene/3d/mesh_instance_3d.cpp#L94

**Comment by AThousandShips** [comment]
This specific code has been in Godot ever since it became open source, unsure if there's a specific reason for it but it's very old and hasn't been changed so far, so hasn't been a major issue it seems

**Comment by Musashi98** [comment]
> This specific code has been in Godot ever since it became open source, unsure if there's a specific reason for it but it's very old and hasn't been changed so far, so hasn't been a major issue it seems

It's not a major issue, it is a tinny thing that can be fixed to improve by a tinny bit the user's experience. I'm really gratefull with the Godot engine community and know better problems are being fixed and new functionallity is being added.

**Comment by TokageItLab** [comment]
I assume it is sorted because hashmaps are not uniquely ordered, so the order may change each time a scene is re-imported.

I suppose we could implement a function like `set_blend_shape_value_by_name()` as a wrapper function of `set_blend_shape_value(find_blend_shape_by_name())`, but string comparisons simply have poor performance, so it is wiser to use indexes in the program. We recommend that users cache and use the indexes obtained with `find_blend_shape_by_name()` generally.

**Comment by docfail** [comment]
> I assume it is sorted because hashmaps are not uniquely ordered since if not so the order may change each time a scene is re-imported. Perhaps this index may be not consistent within glTF in the past; I remember blender glTF Exporter had such a problem in bone/bone-attachment before ([#61641](https://github.com/godotengine/godot/issues/61641)). But I am not sure what is the state for the blendshape now. Thought? [@fire](https://github.com/fire)
> 
> I suppose we could implement a function like `set_blend_shape_value_by_name()` as a wrapper function of `set_blend_shape_value(find_blend_shape_by_name())`, but string comparisons simply have poor performance, so it is wiser to use indexes in the program. We recommend that users cache and use the indexes obtained with `find_blend_shape_by_name()` generally.

I think in the interest of a better user experience, a new method, or rename of the existing method might be beneficial. The fact that the blendshapes are in what is effectively arbitrary order (because realistically, noone is ordering their morphtargets/blendshapes alphabetically in their modeling programs, why would they?) necessitates the use of the find_blend_shape_by_name() method nearly 100% of the time. 
This wouldn't be a big deal, but unfortunately most of the godot documentation is quite hard to google, so most users are likely to find the get/set blend shape methods by autocompleting in the editor. They are not _nearly_ as likely to discover the "find" blend shape method however, because despite it being named well from a pure descriptive perspective, it doesn't begin with "get", which is generally the expected naming pattern for things that aren't expected to be a "search" (which getting the index of a blendshape by its name is technically, but noone who didn't write the method would think of it that way). 
If it was obvious that this was the pattern expected, and/or the method was named in a way that made it easy to discover, I'm willing to bet the original poster of this wouldn't have even made the post. 
I myself only know that the find_blend_shape_by_name() method exists precisely because I found this post and you mentioned it (thank you by the way, you saved me a huge headache), as google sends you here instead of to a useful documentation page.

**Comment by Musashi98** [comment]
> I assume it is sorted because hashmaps are not uniquely ordered since if not so the order may change each time a scene is re-imported. Perhaps this index may be not consistent within glTF in the past; I remember blender glTF Exporter had such a problem in bone/bone-attachment before ([#61641](https://github.com/godotengine/godot/issues/61641)). But I am not sure what is the state for the blendshape now. Thought? [@fire](https://github.com/fire)
> 
> I suppose we could implement a function like `set_blend_shape_value_by_name()` as a wrapper function of `set_blend_shape_value(find_blend_shape_by_name())`, but string comparisons simply have poor performance, so it is wiser to use indexes in the program. We recommend that users cache and use the indexes obtained with `find_blend_shape_by_name()` generally.

I agree caching the index is the best way to go for now, thank you very much! In the future would be nice Godot implemented this wrapper you propossed (set_blend_shape_value_by_name) and auto cache the index.
